organization-mgt-authz-valve/src/main/java/org/wso2/carbon/identity/organization/mgt/authz/valve/OrganizationMgtAuthzValve.java

```diff
@@ -119,16 +121,38 @@ public void invoke(Request request, Response response) throws IOException, Servl
                 if (authorizationResult.getAuthorizationStatus().equals(AuthorizationStatus.GRANT)) {
                     getNext().invoke(request, response);
                 } else {
-                    handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_FORBIDDEN);
+                    throw new AuthzServiceServerException(ERROR_NOT_AUTHORIZED.getCode());
                 }
             } catch (AuthzServiceServerException e) {
-                handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_BAD_REQUEST);
+                if (StringUtils.equalsIgnoreCase(e.getMessage(), ERROR_NOT_AUTHORIZED.getCode())) {
+                    handleErrorResponse(authenticationContext, response, MapErrorCodes(ERROR_NOT_AUTHORIZED.getCode()));
+                } else {
+                    handleErrorResponse(authenticationContext, response, HttpServletResponse.SC_BAD_REQUEST);
+                }
             }
         } else {
             getNext().invoke(request, response);
         }
     }
 
+    /*
+    * @param code Error code
+    * @return error code
+    * */
+    private int MapErrorCodes(String code) {
```

Comment:
Method name should be in Camel case.

Reply:
fixed in 09a513ada49f10161237ca15321a9a9d6fc2efd4